### ISSUE TITLE
[FEATURE]:  Support more dataset and custom special token for reward-model training

### ISSUE THREAD

**Issue by thaumstrial** [hpcaitech/ColossalAI#3097]
### Describe the feature

The following datasets were not found to be supported in the [readme ](https://github.com/hpcaitech/ColossalAI/tree/main/applications/ChatGPT/examples)for training the reward model

[openai/summarize_from_feedback](https://huggingface.co/datasets/openai/summarize_from_feedback)
 [openai/webgpt_comparisons](https://huggingface.co/datasets/openai/webgpt_comparisons)
 [Dahoas/instruct-synthetic-prompt-responses](https://huggingface.co/datasets/Dahoas/instruct-synthetic-prompt-responses)

Are there any plans for that?
I have implemented these locally, whether the project needs this code or not?

And at the [reward_dataset](https://github.com/hpcaitech/ColossalAI/blob/main/applications/ChatGPT/chatgpt/dataset/reward_dataset.py)
Special tokens are directly added to the code, which lacks generality. We may can provide custom added special tokens to adapt different tokenizers.

**Comment by thaumstrial** [comment]
btw, it's better to random shuffle the dataset or models will be overfitting.

**Comment by ht-zhou** [comment]
Thanks for your feedback.We have changed our rm training code a lot last week, and it will be release soon in this week.

**Comment by ht-zhou** [comment]
> btw, it's better to random shuffle the dataset or models will be overfitting.

The rm finetuning on these datasets is set to 1 epoch refer to rlhf papers.
Anyway, shuffling dataset is necessary, and we will fix it in our coming pr.

**Comment by ht-zhou** [comment]
> ### Describe the feature
> The following datasets were not found to be supported in the [readme ](https://github.com/hpcaitech/ColossalAI/tree/main/applications/ChatGPT/examples)for training the reward model
> 
> [openai/summarize_from_feedback](https://huggingface.co/datasets/openai/summarize_from_feedback) [openai/webgpt_comparisons](https://huggingface.co/datasets/openai/webgpt_comparisons) [Dahoas/instruct-synthetic-prompt-responses](https://huggingface.co/datasets/Dahoas/instruct-synthetic-prompt-responses)
> 
> Are there any plans for that? I have implemented these locally, whether the project needs this code or not?
> 
> And at the [reward_dataset](https://github.com/hpcaitech/ColossalAI/blob/main/applications/ChatGPT/chatgpt/dataset/reward_dataset.py) Special tokens are directly added to the code, which lacks generality. We may can provide custom added special tokens to adapt different tokenizers.

And also we will support these datasets soon.

**Comment by zhohuiluo** [comment]
@thaumstrial Could you share your code?，thank you very much，the reward dataset seems not fit task excluding chat

**Comment by binmakeswell** [comment]
We will further update it recently, welcome to stay tuned!
